### ISSUE TITLE
Crash in raft_member_t

### ISSUE THREAD

**Issue by timmaxw** [rethinkdb/rethinkdb#4369]
This came up while shutting down an 8-server cluster:

```
error: Error in ../src/containers/scoped.hpp at line 61:
error: Assertion failed: [ptr_ == __null] 
error: Backtrace:
error: Tue Jun  9 18:41:15 2015
       
       1: rethinkdb_backtrace(void**, int) at rethinkdb_backtrace.cc:100
       2: backtrace_t::backtrace_t() at backtrace.cc:203
       3: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at backtrace.cc:283
       4: format_backtrace(bool) at backtrace.cc:198
       5: report_fatal_error(char const*, int, char const*, ...) at errors.cc:83
       6: scoped_ptr_t<watchdog_timer_t::blocker_t>::init(watchdog_timer_t::blocker_t*) at scoped.hpp:61
       7: raft_member_t<table_raft_state_t>::on_connected_members_change(raft_member_id_t const&, boost::optional<unsigned long> const*)::{lambda()#1}::operator()() const at raft_core.tcc:872
       8: callable_action_instance_t<raft_member_t<table_raft_state_t>::on_connected_members_change(raft_member_id_t const&, boost::optional<unsigned long> const*)::{lambda()#1}>::run_action() at callable_action.hpp:31
       9: callable_action_wrapper_t::run() at runtime_utils.cc:43
       10: coro_t::run() at coroutines.cc:205
       11: coro_t* coro_t::spawn_sometime<raft_member_t<table_raft_state_t>::on_connected_members_change(raft_member_id_t const&, boost::optional<unsigned long> const*)::{lambda()#1}>(raft_member_t<table_raft_state_t>::on_connected_members_change(raft_member_id_t const&, boost::optional<unsigned long> const*)::{lambda()#1}&&) at coroutines.hpp:59
       12: raft_member_t<table_raft_state_t>::on_connected_members_change(raft_member_id_t const&, boost::optional<unsigned long> const*) at raft_core.tcc:842
       13: std::_Mem_fn<void (raft_member_t<table_raft_state_t>::*)(raft_member_id_t const&, boost::optional<unsigned long> const*)>::operator()(raft_member_t<table_raft_state_t>*, raft_member_id_t const&, boost::optional<unsigned long> const*) const at functional:551
       14: void std::_Bind<std::_Mem_fn<void (raft_member_t<table_raft_state_t>::*)(raft_member_id_t const&, boost::optional<unsigned long> const*)> (raft_member_t<table_raft_state_t>*, std::_Placeholder<1>, std::_Placeholder<2>)>::__call<void, raft_member_id_t const&, boost::optional<unsigned long> const*&&, 0, 1, 2>(std::tuple<raft_member_id_t const&, boost::optional<unsigned long> const*&&>&&, std::_Index_tuple<0, 1, 2>) at functional:1146
       15: void std::_Bind<std::_Mem_fn<void (raft_member_t<table_raft_state_t>::*)(raft_member_id_t const&, boost::optional<unsigned long> const*)> (raft_member_t<table_raft_state_t>*, std::_Placeholder<1>, std::_Placeholder<2>)>::operator()<raft_member_id_t const&, boost::optional<unsigned long> const*, void>(raft_member_id_t const&, boost::optional<unsigned long> const*&&) at functional:1204
       16: std::_Function_handler<void (raft_member_id_t const&, boost::optional<unsigned long> const*), std::_Bind<std::_Mem_fn<void (raft_member_t<table_raft_state_t>::*)(raft_member_id_t const&, boost::optional<unsigned long> const*)> (raft_member_t<table_raft_state_t>*, std::_Placeholder<1>, std::_Placeholder<2>)> >::_M_invoke(std::_Any_data const&, raft_member_id_t const&, boost::optional<unsigned long> const*) at functional:1780
       17: std::function<void (raft_member_id_t const&, boost::optional<unsigned long> const*)>::operator()(raft_member_id_t const&, boost::optional<unsigned long> const*) const at functional:2161
       18: watchable_map_t<raft_member_id_t, boost::optional<unsigned long> >::notify_change(raft_member_id_t const&, boost::optional<unsigned long> const*, rwi_lock_assertion_t::write_acq_t*)::{lambda(std::function<void (raft_member_id_t const&, boost::optional<unsigned long> const*)> const&)#1}::operator()(std::function<void (raft_member_id_t const&, boost::optional<unsigned long> const*)> const&) const at watchable_map.tcc:83
       19: void publisher_controller_t<std::function<void (raft_member_id_t const&, boost::optional<unsigned long> const*)> >::publish<watchable_map_t<raft_member_id_t, boost::optional<unsigned long> >::notify_change(raft_member_id_t const&, boost::optional<unsigned long> const*, rwi_lock_assertion_t::write_acq_t*)::{lambda(std::function<void (raft_member_id_t const&, boost::optional<unsigned long> const*)> const&)#1}>(watchable_map_t<raft_member_id_t, boost::optional<unsigned long> >::notify_change(raft_member_id_t const&, boost::optional<unsigned long> const*, rwi_lock_assertion_t::write_acq_t*)::{lambda(std::function<void (raft_member_id_t const&, boost::optional<unsigned long> const*)> const&)#1} const&) at pubsub.hpp:139
       20: watchable_map_t<raft_member_id_t, boost::optional<unsigned long> >::notify_change(raft_member_id_t const&, boost::optional<unsigned long> const*, rwi_lock_assertion_t::write_acq_t*) at watchable_map.tcc:80
       21: watchable_map_transform_t<raft_member_id_t, raft_business_card_t<table_raft_state_t>, raft_member_id_t, boost::optional<unsigned long> >::do_notify_change(raft_member_id_t const&, boost::optional<unsigned long> const*, rwi_lock_assertion_t::write_acq_t*) at watchable_transform.hpp:55
       22: watchable_map_transform_t<raft_member_id_t, raft_business_card_t<table_raft_state_t>, raft_member_id_t, boost::optional<unsigned long> >::watchable_map_transform_t(watchable_map_t<raft_member_id_t, raft_business_card_t<table_raft_state_t> >*)::{lambda(raft_member_id_t const&, raft_business_card_t<table_raft_state_t> const*)#1}::operator()(raft_member_id_t const&, raft_business_card_t<table_raft_state_t> const*) const at watchable_transform.tcc:19
       23: std::_Function_handler<void (raft_member_id_t const&, raft_business_card_t<table_raft_state_t> const*), watchable_map_transform_t<raft_member_id_t, raft_business_card_t<table_raft_state_t>, raft_member_id_t, boost::optional<unsigned long> >::watchable_map_transform_t(watchable_map_t<raft_member_id_t, raft_business_card_t<table_raft_state_t> >*)::{lambda(raft_member_id_t const&, raft_business_card_t<table_raft_state_t> const*)#1}>::_M_invoke(std::_Any_data const&, raft_member_id_t const&, raft_business_card_t<table_raft_state_t> const*) at functional:1780
       24: std::function<void (raft_member_id_t const&, raft_business_card_t<table_raft_state_t> const*)>::operator()(raft_member_id_t const&, raft_business_card_t<table_raft_state_t> const*) const at functional:2161
       25: watchable_map_t<raft_member_id_t, raft_business_card_t<table_raft_state_t> >::notify_change(raft_member_id_t const&, raft_business_card_t<table_raft_state_t> const*, rwi_lock_assertion_t::write_acq_t*)::{lambda(std::function<void (raft_member_id_t const&, raft_business_card_t<table_raft_state_t> const*)> const&)#1}::operator()(std::function<void (raft_member_id_t const&, raft_business_card_t<table_raft_state_t> const*)> const&) const at watchable_map.tcc:83
       26: void publisher_controller_t<std::function<void (raft_member_id_t const&, raft_business_card_t<table_raft_state_t> const*)> >::publish<watchable_map_t<raft_member_id_t, raft_business_card_t<table_raft_state_t> >::notify_change(raft_member_id_t const&, raft_business_card_t<table_raft_state_t> const*, rwi_lock_assertion_t::write_acq_t*)::{lambda(std::function<void (raft_member_id_t const&, raft_business_card_t<table_raft_state_t> const*)> const&)#1}>(watchable_map_t<raft_member_id_t, raft_business_card_t<table_raft_state_t> >::notify_change(raft_member_id_t const&, raft_business_card_t<table_raft_state_t> const*, rwi_lock_assertion_t::write_acq_t*)::{lambda(std::function<void (raft_member_id_t const&, raft_business_card_t<table_raft_state_t> const*)> const&)#1} const&) at pubsub.hpp:139
       27: watchable_map_t<raft_member_id_t, raft_business_card_t<table_raft_state_t> >::notify_change(raft_member_id_t const&, raft_business_card_t<table_raft_state_t> const*, rwi_lock_assertion_t::write_acq_t*) at watchable_map.tcc:80
       28: watchable_map_var_t<raft_member_id_t, raft_business_card_t<table_raft_state_t> >::entry_t::change(std::function<bool (raft_business_card_t<table_raft_state_t>*)> const&) at watchable_map.tcc:173
       29: watchable_map_keyed_var_t<peer_id_t, raft_member_id_t, raft_business_card_t<table_raft_state_t> >::set_key(peer_id_t const&, raft_member_id_t const&, raft_business_card_t<table_raft_state_t> const&) at watchable_transform.hpp:161
       30: table_manager_t::on_table_directory_change(std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*) at table_manager.cc:206
       31: std::_Mem_fn<void (table_manager_t::*)(std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*)>::operator()(table_manager_t*, std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*) const at functional:551
       32: void std::_Bind<std::_Mem_fn<void (table_manager_t::*)(std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*)> (table_manager_t*, std::_Placeholder<1>, std::_Placeholder<2>)>::__call<void, std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*&&, 0, 1, 2>(std::tuple<std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*&&>&&, std::_Index_tuple<0, 1, 2>) at functional:1146
       33: void std::_Bind<std::_Mem_fn<void (table_manager_t::*)(std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*)> (table_manager_t*, std::_Placeholder<1>, std::_Placeholder<2>)>::operator()<std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*, void>(std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*&&) at functional:1204
       34: std::_Function_handler<void (std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*), std::_Bind<std::_Mem_fn<void (table_manager_t::*)(std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*)> (table_manager_t*, std::_Placeholder<1>, std::_Placeholder<2>)> >::_M_invoke(std::_Any_data const&, std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*) at functional:1780
       35: std::function<void (std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*)>::operator()(std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*) const at functional:2161
       36: watchable_map_t<std::pair<peer_id_t, uuid_u>, table_manager_bcard_t>::notify_change(std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*, rwi_lock_assertion_t::write_acq_t*)::{lambda(std::function<void (std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*)> const&)#1}::operator()(std::function<void (std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*)> const&) const at watchable_map.tcc:83
       37: void publisher_controller_t<std::function<void (std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*)> >::publish<watchable_map_t<std::pair<peer_id_t, uuid_u>, table_manager_bcard_t>::notify_change(std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*, rwi_lock_assertion_t::write_acq_t*)::{lambda(std::function<void (std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*)> const&)#1}>(watchable_map_t<std::pair<peer_id_t, uuid_u>, table_manager_bcard_t>::notify_change(std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*, rwi_lock_assertion_t::write_acq_t*)::{lambda(std::function<void (std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*)> const&)#1} const&) at pubsub.hpp:139
       38: watchable_map_t<std::pair<peer_id_t, uuid_u>, table_manager_bcard_t>::notify_change(std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const*, rwi_lock_assertion_t::write_acq_t*) at watchable_map.tcc:80
       39: watchable_map_var_t<std::pair<peer_id_t, uuid_u>, table_manager_bcard_t>::set_key_no_equals(std::pair<peer_id_t, uuid_u> const&, table_manager_bcard_t const&) at watchable_map.tcc:256
       40: directory_map_read_manager_t<uuid_u, table_manager_bcard_t>::do_update(peer_id_t, auto_drainer_t::lock_t, auto_drainer_t::lock_t, unsigned long, uuid_u const&, boost::optional<table_manager_bcard_t> const&) at map_read_manager.tcc:89
       41: boost::_mfi::mf6<void, directory_map_read_manager_t<uuid_u, table_manager_bcard_t>, peer_id_t, auto_drainer_t::lock_t, auto_drainer_t::lock_t, unsigned long, uuid_u const&, boost::optional<table_manager_bcard_t> const&>::operator()(directory_map_read_manager_t<uuid_u, table_manager_bcard_t>*, peer_id_t, auto_drainer_t::lock_t, auto_drainer_t::lock_t, unsigned long, uuid_u const&, boost::optional<table_manager_bcard_t> const&) const at mem_fn_template.hpp:732
       42: void boost::_bi::list7<boost::_bi::value<directory_map_read_manager_t<uuid_u, table_manager_bcard_t>*>, boost::_bi::value<peer_id_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<unsigned long>, boost::_bi::value<uuid_u>, boost::_bi::value<boost::optional<table_manager_bcard_t> > >::operator()<boost::_mfi::mf6<void, directory_map_read_manager_t<uuid_u, table_manager_bcard_t>, peer_id_t, auto_drainer_t::lock_t, auto_drainer_t::lock_t, unsigned long, uuid_u const&, boost::optional<table_manager_bcard_t> const&>, boost::_bi::list0>(boost::_bi::type<void>, boost::_mfi::mf6<void, directory_map_read_manager_t<uuid_u, table_manager_bcard_t>, peer_id_t, auto_drainer_t::lock_t, auto_drainer_t::lock_t, unsigned long, uuid_u const&, boost::optional<table_manager_bcard_t> const&>&, boost::_bi::list0&, int) at bind.hpp:670
       43: boost::_bi::bind_t<void, boost::_mfi::mf6<void, directory_map_read_manager_t<uuid_u, table_manager_bcard_t>, peer_id_t, auto_drainer_t::lock_t, auto_drainer_t::lock_t, unsigned long, uuid_u const&, boost::optional<table_manager_bcard_t> const&>, boost::_bi::list7<boost::_bi::value<directory_map_read_manager_t<uuid_u, table_manager_bcard_t>*>, boost::_bi::value<peer_id_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<unsigned long>, boost::_bi::value<uuid_u>, boost::_bi::value<boost::optional<table_manager_bcard_t> > > >::operator()() at bind_template.hpp:20
       44: callable_action_instance_t<boost::_bi::bind_t<void, boost::_mfi::mf6<void, directory_map_read_manager_t<uuid_u, table_manager_bcard_t>, peer_id_t, auto_drainer_t::lock_t, auto_drainer_t::lock_t, unsigned long, uuid_u const&, boost::optional<table_manager_bcard_t> const&>, boost::_bi::list7<boost::_bi::value<directory_map_read_manager_t<uuid_u, table_manager_bcard_t>*>, boost::_bi::value<peer_id_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<unsigned long>, boost::_bi::value<uuid_u>, boost::_bi::value<boost::optional<table_manager_bcard_t> > > > >::run_action() at callable_action.hpp:31
       45: callable_action_wrapper_t::run() at runtime_utils.cc:43
       46: coro_t::run() at coroutines.cc:205
       47: coro_t* coro_t::spawn_sometime<boost::_bi::bind_t<void, boost::_mfi::mf6<void, directory_map_read_manager_t<uuid_u, table_manager_bcard_t>, peer_id_t, auto_drainer_t::lock_t, auto_drainer_t::lock_t, unsigned long, uuid_u const&, boost::optional<table_manager_bcard_t> const&>, boost::_bi::list7<boost::_bi::value<directory_map_read_manager_t<uuid_u, table_manager_bcard_t>*>, boost::_bi::value<peer_id_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<unsigned long>, boost::_bi::value<uuid_u>, boost::_bi::value<boost::optional<table_manager_bcard_t> > > > >(boost::_bi::bind_t<void, boost::_mfi::mf6<void, directory_map_read_manager_t<uuid_u, table_manager_bcard_t>, peer_id_t, auto_drainer_t::lock_t, auto_drainer_t::lock_t, unsigned long, uuid_u const&, boost::optional<table_manager_bcard_t> const&>, boost::_bi::list7<boost::_bi::value<directory_map_read_manager_t<uuid_u, table_manager_bcard_t>*>, boost::_bi::value<peer_id_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<auto_drainer_t::lock_t>, boost::_bi::value<unsigned long>, boost::_bi::value<uuid_u>, boost::_bi::value<boost::optional<table_manager_bcard_t> > > >&&) at coroutines.hpp:59
       48: directory_map_read_manager_t<uuid_u, table_manager_bcard_t>::on_message(connectivity_cluster_t::connection_t*, auto_drainer_t::lock_t, read_stream_t*) at map_read_manager.tcc:52
       49: connectivity_cluster_t::run_t::handle(keepalive_tcp_conn_stream_t*, boost::optional<peer_id_t>, boost::optional<peer_address_t>, auto_drainer_t::lock_t, bool*) at cluster.cc:1126
       50: connectivity_cluster_t::run_t::on_new_connection(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t) at cluster.cc:285
       51: std::_Mem_fn<void (connectivity_cluster_t::run_t::*)(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t)>::operator()(connectivity_cluster_t::run_t*, scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t) const at functional:551
       52: void std::_Bind<std::_Mem_fn<void (connectivity_cluster_t::run_t::*)(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t)> (connectivity_cluster_t::run_t*, std::_Placeholder<1>, auto_drainer_t::lock_t)>::__call<void, scoped_ptr_t<linux_tcp_conn_descriptor_t>&, 0, 1, 2>(std::tuple<scoped_ptr_t<linux_tcp_conn_descriptor_t>&>&&, std::_Index_tuple<0, 1, 2>) at functional:1146
       53: void std::_Bind<std::_Mem_fn<void (connectivity_cluster_t::run_t::*)(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t)> (connectivity_cluster_t::run_t*, std::_Placeholder<1>, auto_drainer_t::lock_t)>::operator()<scoped_ptr_t<linux_tcp_conn_descriptor_t>&, void>(scoped_ptr_t<linux_tcp_conn_descriptor_t>&) at functional:1204
       54: std::_Function_handler<void (scoped_ptr_t<linux_tcp_conn_descriptor_t>&), std::_Bind<std::_Mem_fn<void (connectivity_cluster_t::run_t::*)(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t)> (connectivity_cluster_t::run_t*, std::_Placeholder<1>, auto_drainer_t::lock_t)> >::_M_invoke(std::_Any_data const&, scoped_ptr_t<linux_tcp_conn_descriptor_t>&) at functional:1780
       55: std::function<void (scoped_ptr_t<linux_tcp_conn_descriptor_t>&)>::operator()(scoped_ptr_t<linux_tcp_conn_descriptor_t>&) const at functional:2161
       56: linux_nonthrowing_tcp_listener_t::handle(int) at network.cc:924
       57: std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)>::operator()(linux_nonthrowing_tcp_listener_t*, int) const at functional:551
       58: void std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)>::__call<void, , 0, 1>(std::tuple<>&&, std::_Index_tuple<0, 1>) at functional:1146
       59: void std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)>::operator()<, void>() at functional:1204
       60: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)> >::run_action() at callable_action.hpp:31
       61: callable_action_wrapper_t::run() at runtime_utils.cc:43
       62: coro_t::run() at coroutines.cc:205
       63: void coro_t::spawn_now_dangerously<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)> >(std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)>&&) at coroutines.hpp:53
       64: linux_nonthrowing_tcp_listener_t::accept_loop(auto_drainer_t::lock_t) at network.cc:891
       65: std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)>::operator()(linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t) const at functional:551
       66: void std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t)>::__call<void, , 0, 1>(std::tuple<>&&, std::_Index_tuple<0, 1>) at functional:1146
       67: void std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t)>::operator()<, void>() at functional:1204
       68: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t)> >::run_action() at callable_action.hpp:31
       69: callable_action_wrapper_t::run() at runtime_utils.cc:43
       70: coro_t::run() at coroutines.cc:205
       71: coro_t* coro_t::spawn_sometime<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t)> >(std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t)>&&) at coroutines.hpp:59
       72: linux_nonthrowing_tcp_listener_t::begin_listening() at network.cc:684
       73: linux_tcp_listener_t::linux_tcp_listener_t(linux_tcp_bound_socket_t*, std::function<void (scoped_ptr_t<linux_tcp_conn_descriptor_t>&)> const&) at network.cc:966
       74: connectivity_cluster_t::run_t::run_t(connectivity_cluster_t*, std::set<ip_address_t, std::less<ip_address_t>, std::allocator<ip_address_t> > const&, peer_address_t const&, int, int) at cluster.cc:244
error: Exiting.
```

**Comment by danielmewes** [comment]
The `continuous.serial` test that uses a 4-server cluster is currently failing because of this.

**Comment by danielmewes** [comment]
Fix is in CR 2975

**Comment by danielmewes** [comment]
Fixed in `raft` 5ab7736236a0532d0c5867c602941d17e9d03ff1
